drivers/gpio/gpio_npm1300.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nordic_npm1300_gpio
+
+#include <errno.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/dt-bindings/gpio/nordic-npm1300-gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+#include <zephyr/sys/byteorder.h>
+
+/* nPM1300 GPIO base address */
+#define NPM_GPIO_BASE 0x06U
+#define NPM_LED_BASE  0x0AU
+
+/* nPM1300 GPIO registers offsets */
+#define NPM_GPIO_OFFSET_MODE	  0x00U
+#define NPM_GPIO_OFFSET_DRIVE	  0x05U
+#define NPM_GPIO_OFFSET_PULLUP	  0x0AU
+#define NPM_GPIO_OFFSET_PULLDOWN  0x0FU
+#define NPM_GPIO_OFFSET_OPENDRAIN 0x14U
+#define NPM_GPIO_OFFSET_DEBOUNCE  0x19U
+
+/* nPM1300 LED register offsets */
+#define NPM_LED_OFFSET_MODE    0x00U
+#define NPM_LED_OFFSET_DRV_SET 0x03U
+#define NPM_LED_OFFSET_DRV_CLR 0x04U
+
+/* nPM1300 Channel counts */
+#define NPM1300_GPIO_PINS     5U
+#define NPM1300_GPIO_LED_PINS 3U
+
+#define NPM1300_GPIO_GPIINPUT	    0
+#define NPM1300_GPIO_GPILOGIC1	    1
+#define NPM1300_GPIO_GPILOGIC0	    2
+#define NPM1300_GPIO_GPIEVENTRISE   3
+#define NPM1300_GPIO_GPIEVENTFALL   4
+#define NPM1300_GPIO_GPOIRQ	    5
+#define NPM1300_GPIO_GPORESET	    6
+#define NPM1300_GPIO_GPOPWRLOSSWARN 7
+#define NPM1300_GPIO_GPOLOGIC1	    8
+#define NPM1300_GPIO_GPOLOGIC0	    9
+
+#define NPM1300_GPIO_LED_ERROR	  0
+#define NPM1300_GPIO_LED_CHARGING 1
+#define NPM1300_GPIO_LED_HOST	  2
+
+struct gpio_npm1300_config {
+	struct gpio_driver_config common;
+	struct i2c_dt_spec bus;
+};
+
+struct gpio_npm1300_data {
+	struct gpio_driver_data common;
+};
+
+/* Write single register to specified address */
+static int reg_write(const struct device *dev, uint8_t base, uint8_t offset, uint8_t data)
+{
+	const struct gpio_npm1300_config *config = dev->config;
+	uint8_t buff[] = {base, offset, data};
+
+	return i2c_write_dt(&config->bus, buff, sizeof(buff));
+}
+
+static int gpio_npm1300_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	return -ENOTSUP;
```

Comment:
err kind of disappointing input level can't be read. Isn't GPILOGIC0/1 set accordingly?